test/pylib/scylla_cluster.py

```diff
@@ -113,6 +113,61 @@ def make_scylla_conf(workdir: pathlib.Path, host_addr: str, seed_addrs: List[str
     '--abort-on-ebadf', '1'
 ]
 
+# [--smp, 1], [--smp, 2] -> [--smp, 2]
+# [--smp, 1], [--smp] -> [--smp]
+# [--smp, 1], [--smp, __missing__] -> [--smp]
```

Comment:
What's the point of `__missing__` though, if the same behavior can be obtained by simply not providing a value? (like the previous example)

Reply:
The syntax is `--name name val`, we need explicit notion of parameter name and value to be able to merge two parameter sets. `__missing__` gives explicit sign that parameter doesn't have a value, otherwise the next word in the list would be treated as a value. The only case when we doesn't need this is at the end of the list, it's obvious that parameter doesn't have a value there.

Reply:
> The syntax is `--name name val`

Not `--name val`? As in `--smp 2`.

> otherwise the next word in the list would be treated as a value

That depends on whether the next word starts with `-`.

Here: `[--key1, value, --key2]` you specify that `key2` has a missing value.
Here: `[--key1, --key2, value]` you specify that `key1` has a missing value. You know that the next word is not a value because it starts with `--`.

Perhaps it's just simpler to implement with `__missing__`?